server/routes/nl.py

```diff
@@ -56,20 +54,29 @@ def _get_preferred_type(types):
 
 def _maps_place(place_str):
   if place_str.lower() in constants.SPECIAL_PLACE_REPLACEMENTS:
+    logging.info(f"place_str {place_str} matched a special place.")
     place_str = constants.SPECIAL_PLACE_REPLACEMENTS[place_str.lower()]
+    logging.info(f"place_str replaced with: {place_str}")
+
   api_key = current_app.config["MAPS_API_KEY"]
   # Note on 02/15/2023: Maps textsearch API has deprecated the use of
   # `input` as a url param and instead wants `query`.
   # Reference: https://developers.google.com/maps/deprecations#unsupported-place-search-deprecation
-  url_formatted = f"{MAPS_API}query={place_str}&key={api_key}"
+  url_formatted = f"{constants.MAPS_API}query={place_str}&key={api_key}"
   r = requests.get(url_formatted)
   resp = r.json()
 
   # Return the first "political" place found.
   if "results" in resp:
     for res in resp["results"]:
-      if "political" in res["types"]:
+      types_found = set(res["types"])
+
+      if constants.MAPS_GEO_TYPES.intersection(types_found):
         return res
+
+  logging.info(
+      f"Maps API did not find a result of type political. Query URL: {url_formatted}. Response: {resp}"
```

Comment:
nit: But we look for stuff beyond "political"?

Reply:
edited